src/main/java/com/onarandombox/MultiverseCore/commands/SetSpawnCommand.java

```diff
@@ -52,28 +61,57 @@ protected void setWorldSpawn(CommandSender sender) {
             Player p = (Player) sender;
             Location l = p.getLocation();
             World w = p.getWorld();
-            MultiverseWorld foundWorld = this.plugin.getMVWorldManager().getMVWorld(w.getName());
-            if (foundWorld != null) {
-                foundWorld.setSpawnLocation(p.getLocation());
-                BlockSafety bs = this.plugin.getBlockSafety();
-                if (!bs.playerCanSpawnHereSafely(p.getLocation()) && foundWorld.getAdjustSpawn()) {
-                    sender.sendMessage("It looks like that location would normally be unsafe. But I trust you.");
-                    sender.sendMessage("I'm turning off the Safe-T-Teleporter for spawns to this world.");
-                    sender.sendMessage("If you want this turned back on just do:");
-                    sender.sendMessage(ChatColor.AQUA + "/mvm set adjustspawn true " + foundWorld.getAlias());
-                    foundWorld.setAdjustSpawn(false);
-                }
-                sender.sendMessage("Spawn was set to: " + plugin.getLocationManipulation().strCoords(p.getLocation()));
-                if (!plugin.saveWorldConfig()) {
-                    sender.sendMessage(ChatColor.RED + "There was an issue saving worlds.yml!  Your changes will only be temporary!");
-                }
-            } else {
-                w.setSpawnLocation(l.getBlockX(), l.getBlockY(), l.getBlockZ());
-                sender.sendMessage("Multiverse does not know about this world, only X,Y and Z set. Please import it to set the spawn fully (Pitch/Yaws).");
-            }
-
+            setWorldSpawn(sender, w, l);
         } else {
             sender.sendMessage("You cannot use this command from the console.");
```

Comment:
It does already! The console version won't work without any coordinates, obviously, but the (sender instanceof Player) check is in the code path that doesn't use coordinates, now. I agree the wording should be changed however, to "You need to give coordinates when using this command from the console!".

Reply:
Looking at just the diff is a bit confusing, because I moved the main logic block to a new method, that gets called from the various 0/4/6 argument handlers. The diff treats that as a remove/add, which hides the logic behind what's happening. The patched source (https://github.com/BaronyCraft/Multiverse-Core/blob/setspawn-command-with-coords/src/main/java/com/onarandombox/MultiverseCore/commands/SetSpawnCommand.java) is probably easier to read than the patch.